File: 1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int,int>umpp;

        for(int i=0;i<nums.size();i++)
        {
            if(umpp.find(target-nums[i])!=umpp.end())
            {
                return {umpp.find(target-nums[i])->second,i};
            }
            else
            {
                umpp.insert({nums[i],i});
            }
        }
        return {};
    }
};
// @lc code=end

File: 1.two-sum.java
/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start

import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer,Integer>hmpp=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            if(hmpp.containsKey(target-nums[i])){
                return new int[]{hmpp.get(target-nums[i]),i};
            }
            else{
                hmpp.put(nums[i], i);
            }
        }

        return new int[]{-1,-1};
        
    }
}
// @lc code=end

